### DISCUSSION TITLE
Targetting the "General Game Audience"

### DISCUSSION THREAD

**Discussion by TheMikirog** [bombsquad-community/plugin-manager#67] [General]
The way the repository states that mods are supposed to target the "general game audience" is extremely vague to me, since it doesn't really explain which kinds of mods are expected.

- If the mod has to target a typical player, does that mean mods that don't work on other servers when joining them are out of the question? 
- If I make a gameplay mod that is server-side like minigames, is it allowed?
- I assume big gameplay changes or reworks are not allowed?

There needs to be more info about what kinds of mods are allowed.

**Comment by rikkolovescats** [comment]
Hi!

> The way the repository states that mods are supposed to target the "general game audience" is extremely vague to me, since it doesn't really explain which kinds of mods are expected.

Yep, the wording can definitely be improved but I don't think it'll be possible to define a clear boundary between what'll get accepted and what will not be. It mostly means plugins that only target a specific set of users should be avoided.

Such as [sahilp_characters.py](https://github.com/rikkolovescats/sahilp-plugins/blob/main/plugins/sahilp_characters.py), whose mashed up character models only work in sahilp's game server. I don't think it makes sense to have this plugin in the main repo since only players close to this server will only ever have a ping good enough to even bother joining this server in the first place.

I also have few personal plugins. One of which all it does is make the output of `ba.internal.get_game_roster()` a little nicer to look at. It needs to be called from the console manually. I find it useful but It didn't seem like it should be here since a usual player wouldn't find it any useful. However, I feel if such a plugin were to be modified to display an overlay text in the foreground game activity making its reach to android and people who aren't comfy with the console, then it should be nice to have it here in the main repo.

And definitely plugins features that cause or could cause problems for other players (spamming game servers, etc.) will not accepted.

These restrictions may also help to keep everything a bit more maintainable. Such as a scenario where a new game version drops and the author who shared a plugin (that only targets a very small platform of users) to this repo in the past is no longer active, community might not have the motivation to update this plugin as it's not useful to them, and this could start to build unmaintained clutter.

> If the mod has to target a typical player, does that mean mods that don't work on other servers when joining them are out of the question?

I think it's okay to have them since a lot of players play single player mode too. Such as [mood_light.py](https://github.com/bombsquad-community/plugin-manager/blob/main/plugins/utilities/mood_light.py).

> If I make a gameplay mod that is server-side like minigames, is it allowed?

Yes, they are allowed. Even if people don't have LAN friends for multi-player minigames, it seems nice to have one place to find all minigames in case they'd like to test it locally first and then wish to put it up into their dedicated game servers' playlist.

> I assume big gameplay changes or reworks are not allowed?

Happy to have them as well as long as they abide with all above.
